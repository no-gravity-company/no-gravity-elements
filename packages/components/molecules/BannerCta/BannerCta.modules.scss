.bg-cta {
    position: relative;
    display: flex;
    align-items: center;
    justify-content: space-between;
    border-radius: 10px;
    padding: var(--space-xl, 2rem);
    background-image: linear-gradient(to right, #408c82d6, #1c2322cf),
      url('https://s3.eu-north-1.amazonaws.com/cdn.no-gravity.es/webs/fisiofit/service-5.webp');
    background-size: cover;
    background-position: center;
  
    &::before {
      content: url('/assets/svgs/abstract-design.svg');
      position: absolute;
      top: 0;
      left: 0px;
    }
  
    .main-call {
      display: flex;
      flex-direction: column;
      gap: var(--space-s, 1rem);
  
      h5,
      p {
        color: var(--text-color-50, #ffffff);
      }
  
      h5 {
        font-weight: 600;
      }
    }
  }
  
  :host {
    display: block;
    padding: var(--section-padding, 2rem);
  }
  
  @media only screen and (max-width: 768px) {
    .bg-cta {
      flex-direction: column;
      align-items: inherit;
      padding: var(--space-l, 1.5rem);
  
      .main-call {
        h5,
        p {
          text-align: center;
        }
      }
  
      slot[name='button'] {
        margin-top: var(--space-m, 1rem);
        width: 100%;
      }
    }
  }
  