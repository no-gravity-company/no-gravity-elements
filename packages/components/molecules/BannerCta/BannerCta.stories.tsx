import { Meta, StoryObj } from '@storybook/web-components';
import { html } from 'lit-html';

import '@no-gravity-elements/banner-cta';
import '@no-gravity-elements/button';

type CustomArgs = {
  title: string;
  message: string;
  buttonLabel: string;
};

const meta: Meta<CustomArgs> = {
  title: 'molecules/BannerCTA',
  component: 'nge-banner-cta',
  parameters: {
    webComponents: {
      tagName: 'nge-banner-cta',
    },
  },
  argTypes: {
    title: {
      control: 'text',
      description: 'Text content for the title slot.',
      defaultValue: 'Your Custom Title',
    },
    message: {
      control: 'text',
      description: 'Text content for the message slot.',
      defaultValue: 'Your custom call-to-action message.',
    },
    buttonLabel: {
      control: 'text',
      description: 'Label for the button inside the button slot.',
      defaultValue: 'Click Me',
    },
  },
  render: ({ title, message, buttonLabel }) => html`
    <style>
      :root {
        --nge-button-padding: 8px 16px;
        --nge-button-border-radius: 280px;
      }
      h5,
      p {
        margin: 0;
      }
    </style>
    <nge-banner-cta>
      <h5 slot="title">${title}</h5>
      <p slot="message">${message}</p>
      <nge-button slot="button" value="${buttonLabel}"></nge-button>
    </nge-banner-cta>
  `,
};

export default meta;
type Story = StoryObj<CustomArgs>;

export const Default: Story = {
  args: {
    title: 'Your Custom Title',
    message: 'Your custom call-to-action message.',
    buttonLabel: 'Click Me',
  },
};
