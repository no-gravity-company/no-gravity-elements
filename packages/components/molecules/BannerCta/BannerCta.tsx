import { h, FunctionComponent, Fragment } from 'preact';
import './BannerCta.modules.scss';

/**
 * <nge-banner-cta>
 *
 * A web component for a customizable call-to-action section.
 *
 * @element nge-banner-cta
 *
 * @slot title - Slot for the call-to-action title.
 * @slot message - Slot for the call-to-action message.
 * @slot button - Slot for the button element or other actions.
 *
 * @cssproperty --space-xl - Padding for the background container (default: 2rem).
 * @cssproperty --text-color-50 - Text color for the title and message.
 * @cssproperty --section-padding - Padding for the call-to-action section (default: 2rem).
 * @cssproperty --space-s - Gap between title and message (default: 1rem).
 * @cssproperty --space-l - Padding for small screen sizes.
 * @cssproperty --space-m - Margin for the button on small screens.
 *
 * @example
 * <nge-banner-cta>
 *   <h5 slot="title">Your Custom Title</h5>
 *   <p slot="message">Your custom call-to-action message.</p>
 *   <nge-button slot="button" value="Click Me"></nge-button>
 * </nge-banner-cta>
 */
const BannerCta: FunctionComponent = () => {
  return (
    <Fragment>
      <div className="bg-cta">
        <div className="main-call">
          <h5 id="cta-title">
            <slot name="title">Default Title</slot>
          </h5>
          <p>
            <slot name="message">Default call-to-action message.</slot>
          </p>
        </div>
        <slot name="button" />
      </div>
    </Fragment>
  );
};

export default BannerCta;
