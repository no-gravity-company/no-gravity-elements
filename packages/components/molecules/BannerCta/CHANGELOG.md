# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [1.0.2](https://github.com/no-gravity-company/no-gravity-elements/compare/@no-gravity-elements/banner-cta@1.0.1...@no-gravity-elements/banner-cta@1.0.2) (2024-12-16)

### Bug Fixes

- bannerCta ([347366f](https://github.com/no-gravity-company/no-gravity-elements/commit/347366fa053a3dfa89368576c3fb262f58a4d486))

## 1.0.1 (2024-12-15)

**Note:** Version bump only for package @no-gravity-elements/banner-cta
