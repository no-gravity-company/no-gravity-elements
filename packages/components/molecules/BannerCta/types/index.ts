export interface BannerCtaProps {
    x: string;
}
    