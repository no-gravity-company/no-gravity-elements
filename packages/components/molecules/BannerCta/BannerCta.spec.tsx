import { h } from 'preact';
import { shallow } from 'enzyme';

import BannerCta from './BannerCta';

describe('BannerCta', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<BannerCta>Test</BannerCta>);
    expect(wrapper).toMatchSnapshot();
  });
});