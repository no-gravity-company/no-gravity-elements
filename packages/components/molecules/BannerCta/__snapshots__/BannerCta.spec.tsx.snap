// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerCta should match the snapshot 1`] = `
<div
  className="bg-cta"
>
  <div
    className="main-call"
  >
    <h5
      id="cta-title"
    >
      <slot
        name="title"
      >
        Default Title
      </slot>
    </h5>
    <p>
      <slot
        name="message"
      >
        Default call-to-action message.
      </slot>
    </p>
  </div>
  <slot
    name="button"
  />
</div>
`;
